src/main/java/org/spongepowered/api/event/item/inventory/UpdateAnvilEvent.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.item.inventory;
+
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.Transaction;
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.util.annotation.eventgen.GenerateFactoryMethod;
+
+/**
+ * Fires whenever the left and right slots of an anvil are filled and a new result is computed.
+ */
+@GenerateFactoryMethod
+public interface UpdateAnvilEvent extends TargetInventoryEvent, Cancellable {
+
+    /**
+     * Returns the new item name.
+     *
+     * @return The new item name
+     */
+    String getItemName();
+
+    /**
+     * Returns the left input item.
+     *
+     * @return The left input item
+     */
+    ItemStackSnapshot getLeft();
+
+    /**
+     * Returns the right input item.
+     *
+     * @return The right input item
+     */
+    ItemStackSnapshot getRight();
+
+    /**
+     * Returns the transaction for the resulting item.
+     *
+     * @return The transaction for the resulting item.
+     */
+    Transaction<ItemStackSnapshot> getItem();
+
+    /**
+     * Returns the transaction for the repair costs.
+     *
+     * @return The transaction for the repair costs.
+     */
+    Transaction<UpdateAnvilEvent.Cost> getCosts();
+
+    /**
+     * The level and material costs for an anvil usage.
+     */
+    interface Cost extends DataSerializable {
```

Comment:
I would prefer this class to be immutable with a builder. That way is transaction also immutable, so you cannot modify the original costs.

Reply:
How about immutable without a builder like it is now?

Reply:
Oh, I didn't notice that it returns a new object. Javadoc is missing `@return`

Ant it's a common practice to use builders in sponge, not sure if it's worth adding one for this.